### DISCUSSION TITLE
Compaction job failed

### DISCUSSION THREAD

**Discussion by rosh0450** [grafana/mimir#7655] [Help and support] [answered]
`ts=2024-03-19T09:05:07.680196404Z caller=compactor.go:652 level=error component=compactor msg="failed to compact user blocks" user=anonymous err="compaction: group 0@17241709254077376921-merge--1710813600000-1710820800000: upload of 01HSAZ9W2V2HTJ0JK82VZ7C4SV failed: upload chunks: upload file /data/compact/0@17241709254077376921-merge--1710813600000-1710820800000/01HSAZ9W2V2HTJ0JK82VZ7C4SV/chunks/000001 as 01HSAZ9W2V2HTJ0JK82VZ7C4SV/chunks/000001: upload s3 object: <html>\r\n<head><title>413 Request Entity Too Large</title></head>\r\n<body>\r\n<center><h1>413 Request Entity Too Large</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>"`

For some reason the compaction job is failing and the reason it is giving is the "request entity too large".
What can be the solution for the issue. should i schedule compactor to run frequently so that it will get less data every time it runs or is there some other solution??

this is my compactor dashboard
![image](https://github.com/grafana/mimir/assets/59838688/cc57bc63-9c66-4222-baed-c378b02a68f0)

**Comment by pstibrany** [comment]
Hello, it seems that compactor has problem with uploading new block back to object storage. File that's being uploaded is typically about 500 MB big. AWS S3 could handle files like that without any problem, but perhaps there's some proxy or you use different storage (not S3)?

**Reply by rosh0450** [reply]
i am using minio object storage

**Reply by pstibrany** [reply]
Minio documents 5 TiB as limit for single PUT operation: https://min.io/docs/minio/container/operations/checklists/thresholds.html

Is there any webserver in front of Minio (eg. nginx) that could possibly impose different limit?

**Reply by rosh0450** [reply]
i have an ingress of class nginx in front of minio's tenant svc

**Reply by pstibrany** [reply, marked as answer]
Nginx has an option to disable this check (`client_max_body_size`), or at least increase the value. It defaults to "1m". Files in blocks are often bigger (~500 MB for chunks files, index file can be in gigabytes, but that depends on how much data your Mimir is ingesting).

https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size

**Reply by rosh0450** [reply]
thanks
